File: 0599-minimum-index-sum-of-two-lists/0599-minimum-index-sum-of-two-lists.cpp
class Solution {
public:
    vector<string> findRestaurant(vector<string>& list1, vector<string>& list2) {
        unordered_map<string,int> map;
        for(int i=0;i<list2.size();i++){
            map[list2[i]]=i;
        }
        int min=INT_MAX;
        vector<string> list;
        for(int i=0;i<list1.size();i++){
            if (map.find(list1[i])!=map.end()) {
                int j = map[list1[i]];
                int sum = i + j;
                if (sum < min) {
                    min = sum;
                    list.clear();
                    list.push_back(list1[i]);
                } else if (sum == min) {
                    list.push_back(list1[i]);
                }
            }
        }
        vector<string> ans(list.size());
        for(int i=0;i<list.size();i++){
            ans[i]=list[i];
        }
        return ans;
    }
};

File: 0599-minimum-index-sum-of-two-lists/0599-minimum-index-sum-of-two-lists.java
class Solution {
    public String[] findRestaurant(String[] list1, String[] list2) {
        HashMap<String,Integer> map=new HashMap<>();
        for(int i=0;i<list2.length;i++){
            map.put(list2[i],i);
        }
        int min=Integer.MAX_VALUE;
        List<String> list=new ArrayList<>();
        for(int i=0;i<list1.length;i++){
            if (map.containsKey(list1[i])) {
                int j = map.get(list1[i]);
                int sum = i + j;
                if (sum < min) {
                    min = sum;
                    list.clear();
                    list.add(list1[i]);
                } else if (sum == min) {
                    list.add(list1[i]);
                }
            }
        }
        String[] ans=new String[list.size()];
        for(int i=0;i<list.size();i++){
            ans[i]=list.get(i);
        }
        return ans;
    }
}